partiql-planner/src/main/kotlin/org/partiql/planner/PartiQLHeader.kt

```diff
@@ -326,8 +337,29 @@ object PartiQLHeader : Header() {
         )
     }
 
-    // TODO
-    private fun coalesce(): List<FunctionSignature.Scalar> = emptyList()
+    // COALESCE(expression, expression, ... )
+    // Initial implementation of Coalesce.
+    // As the number of parameter can not be pre-determined, we wrap those into a list
+    private fun coalesce(): List<FunctionSignature.Scalar> = listOf(
```

Comment:
Another way to model the control flow functions such as `coalesce`/`nullIf` is to rewrite them using `Case When` during planning. That is in my opinion a better approach.

Reply:
Let's mark that as a TODO for the next milestone. This works well enough for us now.

Reply:
After some consideration, I believe that it is better to have the rewrite implemented now.